src/BlazorWebView/src/SharedSource/StaticContentHotReloadManager.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.JSInterop;
+
+[assembly: MetadataUpdateHandler(typeof(Microsoft.AspNetCore.Components.WebView.StaticContentHotReloadManager))]
+
+namespace Microsoft.AspNetCore.Components.WebView
+{
+	internal static class StaticContentHotReloadManager
+	{
+		private delegate void ContentUpdatedHandler(string assemblyName, string relativePath);
+
+		private readonly static Regex ContentUrlRegex = new Regex("^_content/(?<AssemblyName>[^/]+)/(?<RelativePath>.*)");
+		private static event ContentUpdatedHandler? OnContentUpdated;
+
+		private static string? ApplicationAssemblyName { get; } = Assembly.GetEntryAssembly()?.GetName().Name;
+
+		private static readonly Dictionary<(string? AssemblyName, string RelativePath), (string? ContentType, byte[] Content)> _updatedContent = new()
+		{
+			{ (ApplicationAssemblyName, "_framework/static-content-hot-reload.js"), ("text/javascript", Encoding.UTF8.GetBytes(@"
+	export function notifyCssUpdated() {
+		const allLinkElems = Array.from(document.querySelectorAll('link[rel=stylesheet]'));
+		allLinkElems.forEach(elem => elem.href += '');
+	}
+")) }
+		};
+
+		/// <summary>
+		/// MetadataUpdateHandler event. This is invoked by the hot reload host via reflection.
+		/// </summary>
+		public static void UpdateContent(string assemblyName, string relativePath, byte[] content)
+		{
+			_updatedContent[(assemblyName, relativePath)] = (ContentType: null, Content: content);
+
+			// On some platforms (Android), the information about the application assembly name is lost
+			// at compile-time. As a workaround, we treat the app assembly name as null, and treat all
+			// hot reloaded files as being in both an RCL and the app assembly. This works fine if the
+			// names don't clash, but means that RCL files can override an app file if the name does clash.
+			// TODO: Either have the tooling tell us whether this is the main application project, or
+			//       have MAUI somehow retain the information about application assembly name to runtime.
```

Comment:
Note that I've put in a request to have an extra parameter to give us this information, so I might be able to remove this workaround soon.